src/modules/tickets/TicketRepository.ts

```diff
@@ -0,0 +1,52 @@
+import knex from "knex";
+import { Service } from "typedi";
+import config from "../../../knexfile";
+import { ITicket } from "../../entities/ITicket";
+import { UpdateTicketStatus, updateTicket } from "./dto/TicketRequest";
+
+@Service()
+export class TicketRepository {
+  async getAllTicket(): Promise<ITicket[]> {
+    return await knex(config)("tickets").select("*");
+  }
+
+  async getTicket(id: string): Promise<ITicket> {
+    const response = await knex(config)("tickets")
+      .where({ id: id })
+      .select("*");
+    return response[0];
+  }
+
+  async addTicket(ticket: ITicket): Promise<string> {
+    const response = await knex(config)("tickets").insert({
+      ...ticket,
+      created_at: new Date(),
+      updated_at: new Date(),
+    });
+    return response[0].toString();
```

Comment:
Returning the result of the `insert` operation is not agnostic since it's up to the database driver implementation, don't return the value since it's not implementation agnostic.